### PR TITLE
Fix helm repo constant refreshing

### PR DESCRIPTION
**Pull request by thedadams**
The helm repo was being refreshed because `repoSpec.URL` and `status.URL` were different. They were different because we were ensuring that `status.URL` ended with a '/'. Now we only add the slash if it is needed to download a chart.

The issue is that `base.ResolveReference(u)` will remove the last part of the path unless it ends in a '/'.
You can see the difference in this playground: https://play.golang.org/p/DbXvDC_S6ZX

### CODE REVIEW

**Review by aiyengar2 — APPROVED** [review]

**Comment by cbron** [comment]
https://github.com/rancher/rancher/issues/29557

**Review by cbron — COMMENTED** [review]
lgtm, with note that we might have to fix icon code, waiting for other reviewers too

**Review by cbron — APPROVED** [review]
lgtm, needs a backport after everyone approves

**Review by aiyengar2 — APPROVED** [review]

**Review by rmweir — APPROVED** [review]
LGTM

**Pull request merged**
